api/api.go

```diff
@@ -292,3 +292,44 @@ func (c *Client) Skip(language, slug string) error {
 
 	return errors.New(pe.Error)
 }
+
+// Status sends a request to exercism to fetch the user's
+// completion status for the given language track.
+func (c *Client) Status(language string) ([][]string, error) {
+	url := fmt.Sprintf("%s/tracks/%s/status?key=%s", c.XAPIHost, language, c.APIKey)
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.Do(req, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	defer res.Body.Close()
+	if res.StatusCode == http.StatusNoContent {
+		msg := fmt.Sprintf("You have not yet begun the %v track", language)
+		return nil, errors.New(msg)
+	}
+	//
+	//
+	//
+	//
+	//
+	// API response format
+	// {
+	//   "recent":"leap",
+	//   "submitted":"Jan 1, 2015",
+	//   "fetched":["gigasecond", "raindrops"],
+	//   "skipped":["clock", "poker"]
+	// }
+
+	recent := "leap"
+	submittedOn := "Jan 1, 2015"
+	fetched := []string{"gigasecond", "raindrops"}
+	skipped := []string{"clock", "poker"}
+
+	return [][]string{[]string{recent, submittedOn}, fetched, skipped}, nil
```

Comment:
Is there a particular reason to return a slice of slice here? I think this should just be a custom struct.